addon/components/aria-tabs.js

```diff
@@ -1,6 +1,370 @@
+      const maxTabIndex = tabNodes.length - 1;
+      let newSelectedIndex = null;
+
+      if (selectedIndex != null) {
+        newSelectedIndex = Math.min(selectedIndex, maxTabIndex);
+      } else {
+        newSelectedIndex = defaultIndex || 0;
+      }
+      set(this, 'selectedIndex', newSelectedIndex);
+    }
+  },
+
+  nodesEquals(propName, nodeList) {
+    let previousNodes = get(this, propName);
+    if (previousNodes.length !== nodeList.length) {
+      return false;
+    }
+    let nodeIds = nodeList.map((node) => node.id);
+    let previousNodeIds = previousNodes.map((node) => node.id);
+    return previousNodeIds.every((prevNodeId) => nodeIds.includes(prevNodeId));
+  },
+
+  setSelected(index, event) {
+    // Check index boundary
+    if (index < 0 || index >= get(this, 'tabsNodes.length')) {
+      return;
+    }
+
+    let {
+      onSelect,
+      selectedIndex: last,
+      mode
+    } = getProperties(this, [
+      'onSelect',
+      'selectedIndex',
+      'mode'
+    ]);
+    // Call change event handler
+    if (typeof onSelect === 'function') {
+      if (onSelect(index, last, event) === false) {
+        // Check if the change event handler cancels the tab change
+        return;
+      }
+    }
+
+    set(this, 'focus', event.type === 'keydown');
+
+    if (mode === MODE_UNCONTROLLED) {
+      set(this, 'selectedIndex', index);
+    }
+  },
+
+  getNextTab(index) {
+    let tabsNodes = get(this, 'tabNodes');
+    const count = tabsNodes.length;
+
+    // Look for non-disabled tab from index to the last tab on the right
+    for (let i = index + 1; i < count; i++) {
+      if (!this.isTabDisabled(tabsNodes[i])) {
+        return i;
+      }
+    }
+
+    // If no tab found, continue searching from first on left to index
+    for (let i = 0; i < index; i++) {
+      if (!this.isTabDisabled(tabsNodes[i])) {
+        return i;
+      }
+    }
+
+    // No tabs are disabled, return index
+    return index;
+  },
+
+  getPrevTab(index) {
+    let tabsNodes = get(this, 'tabNodes');
+    let i = index;
+
+    // Look for non-disabled tab from index to first tab on the left
+    while (i--) {
+      if (!this.isTabDisabled(tabsNodes[i])) {
+        return i;
+      }
+    }
+
+    // If no tab found, continue searching from last tab on right to index
+    i = tabsNodes.length;
+    while (i-- > index) {
+      if (!this.isTabDisabled(tabsNodes[i])) {
+        return i;
+      }
+    }
+
+    // No tabs are disabled, return index
+    return index;
+  },
+
+  getFirstTab() {
+    let tabsNodes = get(this, 'tabNodes');
+    const count = tabsNodes.length;
+
+    // Look for non disabled tab from the first tab
+    for (let i = 0; i < count; i++) {
+      if (!this.isTabDisabled(tabsNodes[i])) {
+        return i;
+      }
+    }
+
+    return null;
+  },
+
+  getLastTab() {
+    let tabsNodes = get(this, 'tabNodes');
+    let i = tabsNodes.length;
+
+    // Look for non disabled tab from the last tab
```

Comment:
idem mais avec `return array[array.length - 1] || null;`

Reply:
This has been ported as is from react-tabs
https://github.com/reactjs/react-tabs/blob/master/src/components/UncontrolledTabs.js